### DISCUSSION TITLE
Binding to a DynamicResource from code

### DISCUSSION THREAD

**Discussion by ArchLeaders** [AvaloniaUI/Avalonia#9087] [Q&A] [answered]
Hello.

I'm trying to bind a UIElement property to a `DynamicResource` from code, but I'm having trouble getting the property to update.
```cs
TextBlock titleBlock = new() {
    Text = setting.Folder,
    FontWeight = FontWeight.Bold,
    Foreground = "SystemControlForegroundChromeGrayBrush".GetBrush(),
    Margin = new(8, 30, 0, 12)
};

...

public static Brush? GetBrush(this string brush)
{
    object? obj = null;
    Application.Current?.TryFindResource(brush, out obj);
    return obj is Color color ? new SolidColorBrush(color) : obj is SolidColorBrush solidColorBrush ? solidColorBrush : null;
}
```

I tried just using `Application.Current?.TryFindResource()`, but this did not create a binding. I also found a `DynamicResourceExtension`, but I could not find any documentation on it.

Is there a way to dynamically bind to a `DynamicResource` from code?

**Comment by maxkatz6** [comment, marked as answer]
Indexer syntax supports bindings https://github.com/AvaloniaUI/Avalonia/issues/4616#issuecomment-945048059

https://docs.avaloniaui.net/docs/data-binding/binding-from-code

**Reply by ArchLeaders** [reply]
Awesome, thanks!
